### PR TITLE
Refactor and add improve all visualizations

### PR DESCRIPTION
**Pull request by szymon159**

### CODE REVIEW

File: data_manager.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class DataManager:
+    def __init__(self, train_data_path: str, test_data_path: str, problem_type: ProblemType) -> None:
+        self.__train_data_path = train_data_path
+        self.__test_data_path = test_data_path
+        self.__problem_type = problem_type
+        self.testing_data = None
+        self.training_data = None
+
+    def read_data(self) -> pd.DataFrame:
```
Comment (Michalski-Artur) [inline data_manager.py]: Is it possible to indicate somehow that Tuple with two objects of DataFrame type is returned here?

File: activation_functions.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from math import e
+from cmath import tanh
+
+class ActivationFunctions:
+    @staticmethod
+    def sigmoid(x_value: float, derivative: bool = False) -> float:
+        if derivative:
+            return x_value * (1.0 - x_value)
```
Comment (Michalski-Artur) [inline activation_functions.py:9]: It is quite misleading as derivative of sigmoid function is: f(x) * (1-f(x)) so same parameter is interprted here as either x or f(x). Maybe add a comment about that and we will refactor it if we have some spare time :)

File: main.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    initial_seed = int(sys.argv[10]) or INTIIAL_SEED
+
+    # Read data
+    data_manager = DataManager(train_data_path, test_data_path, problem_type)
+    data_manager.read_data()
+    input_size = data_manager.get_input_size()
+    output_layer_size = data_manager.get_output_layer_size()
+
+    # Create network structure and train it
+    network = Network(input_size, output_layer_size, use_bias, hidden_layers_count, hidden_layer_neurons_count, activation_function, initial_seed, LEARNING_RATE, EPOCH_MAX)
```
Comment (Michalski-Artur) [inline main.py:44]: Maybe `learning rate` and `epoch max` should be also part of input? Similar as it is done for count of neurons in the hidden layer

File: main.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        raise Exception("Not enough arguments")
+
+    # Parse input arguments
+    train_data_path = sys.argv[1]
+    test_data_path = sys.argv[2]
+    problem_type = __parse_problem_type(sys.argv[3])
+    # Below are the arguments that should have default values
+    visualize_epochs = sys.argv[4] = bool(sys.argv[4]) or False
+    hidden_layers_count = int(sys.argv[5]) or HIDDEN_LAYERS_COUNT
+    hidden_layer_neurons_count = int(sys.argv[6]) or HIDDEN_LAYER_NEURONS_COUNT
```
Comment (Michalski-Artur) [inline main.py]: Maybe initialize `hidden_layer_neurons_count` with: `sqrt(n*m)` or `2*N+1` where N -> size of input and M -> size of output. It should probably give better values than arbitral default value :)

**Review by Michalski-Artur — APPROVED** [review]

File: data_manager.py (hunk 1 above)
Comment (szymon159) [reply data_manager.py]: Yup, will do

File: main.py (hunk 3 above)
Comment (szymon159) [reply main.py:44]: It isn't a requirement to parametrize it. Would keep it out of user's control for simiplicity

File: main.py (hunk 4 above)
Comment (szymon159) [reply main.py]: sqrt(n*m) sounds good. Done

File: activation_functions.py (hunk 2 above)
Comment (szymon159) [reply activation_functions.py:9]: Fixed I guess
